runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -905,6 +928,19 @@ public final void notifyCheckpointComplete(long checkpointId) throws Exception {
       bufferingDoFnRunner.checkpointCompleted(checkpointId);
     }
 
+    List<InMemoryBundleFinalizer.Finalization> finalizations =
+        pendingFinalizations.get(checkpointId);
+    if (finalizations != null) {
+      // remove old finalizations except for the current one
+      pendingFinalizations.clear();
```

Comment:
The default is to have just one checkpoint at a given time, but users can configure concurrent checkpointing which can ack checkpoints out of order.

Reply:
I see, I was basing the logic on UnboundedSourceWrapper but I have removed it now.